tools/js/test/audit_version_regression.js

```diff
@@ -0,0 +1,125 @@
+/* eslint-env mocha */
+
+const assert = require('assert')
+
+/**
+ * Gets the chronological index of a version from the versions.json file
+ * Returns -1 if version is not found, or null for non-version strings like "latest"
+ */
+function getVersionIndex (platform, version) {
+  // Skip non-version entries like "latest"
+  if (version === 'latest') return null
+
+  try {
+    const versions = require(`../../../data/${platform}/common/versions.json`)
+    return versions.indexOf(version)
+  } catch (error) {
+    return -1
+  }
+}
+
+describe('audit version regression in dataPaths', function () {
+  const dataPaths = require('../../../data/dataPaths.json')
+
+  const platforms = ['pc', 'bedrock']
+
+  platforms.forEach(function (platform) {
+    if (!dataPaths[platform]) return
+
+    describe(`${platform} version regression`, function () {
+      it('should not have newer versions pointing to older data than previously seen', function () {
+        const platformData = dataPaths[platform]
+
+        // Load the chronological version order from versions.json
+        const versions = require(`../../../data/${platform}/common/versions.json`)
+
+        // Track the highest version index seen for each data type
+        const highestVersionIndexForDataType = {}
+
+        // Iterate through versions in chronological order
+        for (const version of versions) {
```

Comment:
I don't understand this outer loop. Why not just iterate over dataPaths? You can skip specific file entries like "protocol" if they contain a version path that is not a version